Docs: Add a warning about static queries (#25929)
gatsbyjs/gatsby: docs/docs/conditional-page-builds.md

For more info on the standard build process, please see the [overview of the Gatsby build process](/docs/overview-of-the-gatsby-build-process/).

> > ❗ Static queries are currently not stable to use with conditional page builds. Any query result differences will not trigger pages to rebuild.

Reviewer: I think the issue is that Conditional Page Builds don't account for static queries at the moment. Let's reword this to reflect the same?

> > ❗ Static queries are currently not stable to use with conditional page builds. Any query result differences will not trigger pages to rebuild.

Author: ```suggestion
> ❗ Conditional page builds do not currently account for [static queries](https://www.gatsbyjs.org/docs/static-query/). Any query result differences will not trigger pages to rebuild.
```
